File: backend/internal/models/application.go
package models

import "time"

type Application struct {
	ID          uint            `gorm:"primaryKey" json:"id"`
	JobID       uint            `gorm:"not null;uniqueIndex:idx_job_student" json:"job_id"`
	Job         *Job            `gorm:"foreignKey:JobID" json:"job,omitempty"`
	StudentID   uint            `gorm:"not null;uniqueIndex:idx_job_student" json:"student_id"`
	Student     *StudentProfile `gorm:"foreignKey:StudentID" json:"student,omitempty"`
	CoverLetter string          `gorm:"type:text" json:"cover_letter"`
	Status      string          `gorm:"default:pending;index" json:"status"` // pending | viewed | accepted | rejected
	CreatedAt   time.Time       `json:"created_at"`
}

File: frontend/src/types/application.ts
import type { StudentProfile } from "./user";
import type { Job } from "./job";

export interface Application {
  id: number;
  job_id: number;
  job?: Job;
  student_id: number;
  student?: StudentProfile;
  cover_letter: string;
  status: "pending" | "viewed" | "accepted" | "rejected";
  created_at: string;
}

export interface Notification {
  id: number;
  user_id: number;
  title: string;
  message: string;
  type: "new_job" | "application_update" | "system";
  is_read: boolean;
  link: string;
  created_at: string;
}
